src/pixbuf.c

```diff
@@ -22,26 +22,35 @@ static void write_file(GdkPixbuf *pixbuf, char *path) {
   }
 }
 
-void pixbuf_save_state_to_folder(GdkPixbuf *pixbuf, char *folder,
-                                 char *filename_format) {
+char *format_filename(char *filename_format) {
   time_t current_time = time(NULL);
   char *c_time_string;
   char filename[255];
-  char path[MAX_PATH];
-  size_t bytes_formated;
+  size_t bytes_formatted;
 
   c_time_string = ctime(&current_time);
   c_time_string[strlen(c_time_string) - 1] = '\0';
-  bytes_formated = strftime(filename, sizeof(filename), filename_format,
-                            localtime(&current_time));
-  if (!bytes_formated) {
+  bytes_formatted = strftime(filename, sizeof(filename), filename_format,
+                             localtime(&current_time));
+  if (!bytes_formatted) {
     g_warning(
         "filename_format: %s overflows filename limit - file cannot be saved",
         filename_format);
-    return;
+    return NULL;
   }
 
+  return g_strdup(filename);
+}
+
+void pixbuf_save_state_to_folder(GdkPixbuf *pixbuf, char *folder,
+                                 char *filename_format) {
+  char path[MAX_PATH];
+  char *filename;
+  filename = format_filename(filename_format);
+  if (filename == NULL) return;
+
   g_snprintf(path, MAX_PATH, "%s/%s", folder, filename);
```

Comment:
I think in this case we actually should abort if this fails. We do not want to write to a truncated filename as this could potentially overwrite some file the user did not intend to overwrite.

Reply:
thanks for taking a look at the code!
I hope I get around to taking another look at this at the weekend but I have lots of other things on my mind currently.